starboard/android/shared/media_decoder.cc

```diff
@@ -317,18 +317,18 @@ void MediaDecoder::DecoderThreadFunc() {
                                   &dequeue_output_results);
       }
 
-      if (!tunnel_mode_enabled_) {
-        // Output is only processed when tunnel mode is disabled.
-        if (!dequeue_output_results.empty()) {
-          auto& dequeue_output_result = dequeue_output_results.front();
-          if (dequeue_output_result.index < 0) {
-            host_->RefreshOutputFormat(media_codec_bridge_.get());
-          } else {
-            host_->ProcessOutputBuffer(media_codec_bridge_.get(),
-                                       dequeue_output_result);
-          }
-          dequeue_output_results.erase(dequeue_output_results.begin());
+      if (!dequeue_output_results.empty()) {
+        auto& dequeue_output_result = dequeue_output_results.front();
+        if (dequeue_output_result.index < 0) {
+          host_->RefreshOutputFormat(media_codec_bridge_.get());
+        } else {
+          SB_DCHECK(!tunnel_mode_enabled_);
+          host_->ProcessOutputBuffer(media_codec_bridge_.get(),
+                                     dequeue_output_result);
         }
```

Comment:
Can you rebase the pr and see if any code here is modified?

Reply:
Thanks for the review.
According to Xiaoming's comment, I think maybe we don't need to enable the tunnel mode forcibly.
If we need to turn tunnel mode on the device for experiment, we will let you know first through email.